components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -825,9 +825,18 @@ abstract class AbstractFetchDownloadService : Service() {
     @Suppress("Deprecation")
     @VisibleForTesting
     internal fun useFileStreamLegacy(download: DownloadState, append: Boolean, block: (OutputStream) -> Unit) {
+        createDirectoryIfNoExists(download)
         FileOutputStream(File(download.filePath), append).use(block)
     }
 
+    @VisibleForTesting
+    internal fun createDirectoryIfNoExists(download: DownloadState) {
```

Comment:
nit: if not exists, but better `createDirectoryIfNeeded` ?

Reply:
👍🏽